typed/helpers.go

```diff
@@ -116,12 +97,12 @@ func resolveSchema(s *schema.Schema, tr schema.TypeRef, v *value.Value, ah atomH
 
 func deduceAtom(a schema.Atom, v *value.Value) schema.Atom {
```

Comment:
Can you add a comment explaining what this function is doing? It looks like it usually completely ignores the input schema, how can that be correct?